Add inspiredminds/contao-news-facebook extension (#209)
contao/docs: docs/manual/extensions/news-facebook-sync.en.md

        "contao/conflicts": "@dev",
        "contao/manager-bundle": "4.9.*",
        "inspiredminds/contao-news-facebook": "^6.0"
    },
    "conflict": {
        "contao-components/installer": "<1.3"
    },
    "extra": {
        "contao-component-dir": "assets",
        "symfony": {
            "require": "^4.2"
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    },
    "repositories": [
        {
            "type": "composer",
            "url": "https://token:<YOUR_TOKEN>@packdis.inspiredminds.at/r"
        }
    ]
}
```
{{% /expand %}}

After making this adjustment, run a `composer update` on the command line or update your packages via the Contao 
Manager. Then open the Contao Install Tool to update the database as usual.

## Configuration

To configure this extension, a "Facebook App" needs to created first. The details of this app are then needed for the

> confiugration in the back end.

Reviewer: ```suggestion
configuration in the back end.
```

### Creating a Facebook App

1. First you need to go to [developers.facebook.com](https://developers.facebook.com). If you do not have a Facebook 
developer account yet, you need to create one (or unlock your existing Facebook user as a developer account).

> 2. Under _My Apps_ klick on _Create App_.

Reviewer: ```suggestion
2. Under _My Apps_ click on _Create App_.
```

website. Then click on _Save Changes_.
7. On the left, click on _Settings_ » _Basic_ and enter your website's domain unter _App Domains_. Then click on
_Save Changes_.

{{% notice info %}}
The account under which you create this App must also have the rights to create timeline posts on
the Facebook page you wish to synchronize with (optional if you only want to fetch Facebook Page posts). Alternatively
you can also add further Administrator or Developers after you created the App in the _Roles_ section.
{{% /notice %}}

### Configure the App ID and App Secret in Contao

In Contao, go to _System - Settings_, scroll down to the _Facebook App_ section, enter the __App ID__ and __App Secret__ 
of your Facebook App. You can find this information in your Facebook App under _Settings_ » _Basic_.

![Facebook App settings](/de/extensions/images/en/contao-news-facebook_app_settings_en.png?classes=shadow)

### Configure the Contao News Archive

Go to the settings of your news archive. In the _Facebook sync_ section enable __Facebook sync__ and enter the _numeric_ 
ID of the Facebook Page you want to synchronize with (you can find the ID in the "About" tab of the page or via 

> [findmyfbid.com](http://findmyfbid.com/)). If you wish to automatically fetch the Facebook Page posts and add them to 

Reviewer: ```suggestion
[findmyfbid.com](https://findmyfbid.com/)). If you wish to automatically fetch the Facebook Page posts and add them to 
```

> your news archive, enable the __Fetch page posts__ option. Optionally also define a __Page post date limit__.

Reviewer: ```suggestion
your news archive, enable the __Fetch page posts__ option. Optionally, also define a __Page post date limit__.
```

> You can also do the same for a Facebook Group, if you want to fetch posts from that group. Note however that publishing

Reviewer: ```suggestion
You can also do the same for a Facebook Group, if you want to fetch posts from that group. Note, however, that publishing
```

![Back end settings](/de/extensions/images/en/contao-news-facebook_backend_settings_en.png?classes=shadow)

If you do not use a hook, the default headline length can be configured via

```php
$GLOBALS['FACEBOOK_TITLE_LENGTH'] = …;
```

in your own `config.php`.

## Usage

### Fetch Facebook Page/Group posts

Once you have configured the news archive and also enabled the __Fetch page posts__ or __Fetch group posts__ option the extension will check for new Facebook posts hourly via Contao's cronjob.

### Publish Facebook posts

Contao news entries will be published as Facebook page posts under two conditions:

- in the news entry's properties, under the _Facebook sync_ section, the __Post to Facebook page__ option must be enabled
- the news entry must be published

You can also define a custom message for the Facebook post. If no such message is defined, the news entry's teaser is used by default.

The extension will check for new news entries to be published minutely via Contao's cronjob.

![News settings](/de/extensions/images/en/contao-news-facebook_news_settings_en.png?classes=shadow)

### Manual sync trigger

> There is a button to trigger a sync in the backend, within the global operations for news archives.

Reviewer: ```suggestion
There is a button to trigger a sync in the back end, within the global operations for news archives.
```
